src/dials/algorithms/refinement/refiner.py

```diff
@@ -261,22 +261,41 @@ def _filter_reflections(reflections):
 
     @classmethod
     def from_parameters_data_experiments(cls, params, reflections, experiments):
-        # TODO Checks on the input
-        # E.g. does every experiment contain at least one overlapping model with at
-        # least one other experiment? Are all the experiments either rotation series
-        # or stills (the combination of both not yet supported)?
 
         # copy the experiments
         experiments = _copy_experiments_for_refining(experiments)
 
         # copy and filter the reflections
         reflections = cls._filter_reflections(reflections)
 
-        return cls._build_components(params, reflections, experiments)
+        (
+            experiments,
+            refman,
+            ref_predictor,
+            do_stills,
+        ) = cls._build_reflection_manager_and_predictor(
+            params, reflections, experiments
+        )
+
+        return cls._build_refiner(params, experiments, refman, ref_predictor, do_stills)
 
     @classmethod
-    def _build_components(cls, params, reflections, experiments):
-        """low level build"""
+    def reflections_after_outlier_rejection(cls, params, reflections, experiments):
+
+        # copy the experiments
+        experiments = _copy_experiments_for_refining(experiments)
```

Comment:
Should we add a test case for this which will exercise the code? I have one which is modest in size and runs in < 2 minutes